internal/resources/policy/type/custom/spec_schema.go

```diff
@@ -0,0 +1,229 @@
+
+		err = tmcBlockNodeportServiceRecipeInput.UnmarshalBinary(byteSlice)
+		if err != nil {
+			return data
+		}
+
+		inputRecipeData = &inputRecipe{
+			recipe:                       tmcBlockNodeportServiceRecipe,
+			inputTMCBlockNodeportService: &tmcBlockNodeportServiceRecipeInput,
+		}
+	case string(tmcBlockResourcesRecipe):
+		var tmcBlockResourcesRecipeInput policyrecipecustommodel.VmwareTanzuManageV1alpha1CommonPolicySpecCustomV1TMCCommonRecipe
+
+		err = tmcBlockResourcesRecipeInput.UnmarshalBinary(byteSlice)
+		if err != nil {
+			return data
+		}
+
+		inputRecipeData = &inputRecipe{
+			recipe:                 tmcBlockResourcesRecipe,
+			inputTMCBlockResources: &tmcBlockResourcesRecipeInput,
+		}
+	case string(tmcBlockRolebindingSubjectsRecipe):
+		var tmcBlockRolebindingSubjectsRecipeInput policyrecipecustommodel.VmwareTanzuManageV1alpha1CommonPolicySpecCustomV1TMCBlockRoleBindingSubjects
+
+		err = tmcBlockRolebindingSubjectsRecipeInput.UnmarshalBinary(byteSlice)
+		if err != nil {
+			return data
+		}
+
+		inputRecipeData = &inputRecipe{
+			recipe:                           tmcBlockRolebindingSubjectsRecipe,
+			inputTMCBlockRolebindingSubjects: &tmcBlockRolebindingSubjectsRecipeInput,
+		}
+	case string(tmcExternalIPSRecipe):
+		var tmcExternalIPSRecipeInput policyrecipecustommodel.VmwareTanzuManageV1alpha1CommonPolicySpecCustomV1TMCExternalIPS
+
+		err = tmcExternalIPSRecipeInput.UnmarshalBinary(byteSlice)
+		if err != nil {
+			return data
+		}
+
+		inputRecipeData = &inputRecipe{
+			recipe:              tmcExternalIPSRecipe,
+			inputTMCExternalIps: &tmcExternalIPSRecipeInput,
+		}
+	case string(tmcHTTPSIngressRecipe):
+		var tmcHTTPSIngressRecipeInput policyrecipecustommodel.VmwareTanzuManageV1alpha1CommonPolicySpecCustomV1TMCCommonRecipe
+
+		err = tmcHTTPSIngressRecipeInput.UnmarshalBinary(byteSlice)
+		if err != nil {
+			return data
+		}
+
+		inputRecipeData = &inputRecipe{
+			recipe:               tmcHTTPSIngressRecipe,
+			inputTMCHTTPSIngress: &tmcHTTPSIngressRecipeInput,
+		}
+	case string(tmcRequireLabelsRecipe):
+		var tmcRequireLabelsRecipeInput policyrecipecustommodel.VmwareTanzuManageV1alpha1CommonPolicySpecCustomV1TMCRequireLabels
+
+		err = tmcRequireLabelsRecipeInput.UnmarshalBinary(byteSlice)
+		if err != nil {
+			return data
+		}
+
+		inputRecipeData = &inputRecipe{
+			recipe:                tmcRequireLabelsRecipe,
+			inputTMCRequireLabels: &tmcRequireLabelsRecipeInput,
+		}
+	case string(unknownRecipe):
+		fmt.Printf("[ERROR]: No valid input recipe block found: minimum one valid input recipe block is required among: %v. Please check the schema.", strings.Join(recipesAllowed[:], `, `))
+	}
+
+	flattenSpecData[inputKey] = flattenInput(inputRecipeData)
+
+	if spec.NamespaceSelector != nil {
+		flattenSpecData[policy.NamespaceSelectorKey] = policy.FlattenNamespaceSelector(spec.NamespaceSelector)
+	}
+
+	return []interface{}{flattenSpecData}
+}
+
+func hasSpecChanged(d *schema.ResourceData) bool {
```

Comment:
I think this can be reused across custom and security policies; Please reuse whatever is possible to be reused.